# بسم الله الرحمن الرحيم

In [13]:
from utils import *
import pandas as pd 
from sklearn.model_selection import train_test_split


## Reading the Dataset, cleaning , preparing it for the training 

In [14]:
df = pd.read_csv("dataset/body_level_classification_train.csv")

In [15]:
df_original= cleaning_data(df).copy()

In [16]:
all_class_train,all_class_test=data_spliting(df_original,1/5)

In [17]:
oversampled= over_sampling(all_class_train,sampling_ratio=0.5)

In [18]:
X_train=oversampled.drop(["Body_Level"],axis=1).to_numpy()
Y_train=oversampled["Body_Level"].to_numpy()
X_test =all_class_test.drop(["Body_Level"],axis=1).to_numpy()
Y_test =all_class_test["Body_Level"].to_numpy()

In [19]:
X_train, scaler= data_scaling(X_train)

In [20]:
X_val, X_test, Y_val, Y_test = train_test_split(X_test
                                                     , Y_test, test_size=0.5, random_state=42)

In [21]:
X_val = scaler.transform(X_val)
X_test=scaler.transform(X_test)

In [22]:
model=build_model(2,0.01)

In [23]:
history = model.fit(X_train,Y_train,validation_data=[X_val,Y_val] 
             ,batch_size=32, epochs=400 , verbose =1, 
                       callbacks=[CustomCallback("NNModels/after_sampling/fifth_model.h5")])

Epoch 1/400
46/46 [==============================] - 2s 7ms/step - loss: 0.7513 - accuracy: 0.4946 - val_loss: 0.3606 - val_accuracy: 0.6463
Epoch 2/400
46/46 [==============================] - 0s 3ms/step - loss: 0.3778 - accuracy: 0.6721 - val_loss: 0.2645 - val_accuracy: 0.7619
Epoch 3/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2931 - accuracy: 0.7435 - val_loss: 0.2330 - val_accuracy: 0.8095
Epoch 4/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2557 - accuracy: 0.7748 - val_loss: 0.2100 - val_accuracy: 0.8571
Epoch 5/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2315 - accuracy: 0.7844 - val_loss: 0.1915 - val_accuracy: 0.8844
Epoch 6/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2110 - accuracy: 0.8224 - val_loss: 0.1851 - val_accuracy: 0.8639
Epoch 7/400
46/46 [==============================] - 0s 3ms/step - loss: 0.1958 - accuracy: 0.8401 - val_loss: 0.1672 - val_accuracy: 0.8571
Epoch 8/400
4

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:23        17192
config.json                                    2023-05-15 07:52:23         1588
metadata.json                                  2023-05-15 07:52:23           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0516 - accuracy: 0.9646 - val_loss: 0.0920 - val_accuracy: 0.9728
Epoch 46/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0506 - accuracy: 0.9626 - val_loss: 0.0925 - val_accuracy: 0.9796
Epoch 47/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0500 - accuracy: 0.9633 - val_loss: 0.0952 - val_accuracy: 0.9796
Epoch 48/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0482 - accuracy: 0.9694 - val_loss: 0.0853 - val_accuracy: 0.9796
Epoch 49/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0489 - accuracy: 0.9612 - val_loss: 0.0927 - val_

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:26        17192
config.json                                    2023-05-15 07:52:26         1588
metadata.json                                  2023-05-15 07:52:26           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0359 - accuracy: 0.9769 - val_loss: 0.0948 - val_accuracy: 0.9796
Epoch 69/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0354 - accuracy: 0.9721 - val_loss: 0.0925 - val_accuracy: 0.9728
Epoch 70/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0358 - accuracy: 0.9687 - val_loss: 0.0887 - val_accuracy: 0.9796
Epoch 71/400
34/46 [=====================>........] - ETA: 0s - loss: 0.0332 - accuracy: 0.9807
Yes You are here 0.9768707752227783 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0


File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:27        17192
config.json                                    2023-05-15 07:52:27         1588
metadata.json                                  2023-05-15 07:52:27           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0311 - accuracy: 0.9755 - val_loss: 0.0933 - val_accuracy: 0.9796
Epoch 81/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0299 - accuracy: 0.9781
Yes You are here 0.976190447807312 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name               

46/46 [==============================] - 0s 3ms/step - loss: 0.0281 - accuracy: 0.9762 - val_loss: 0.0949 - val_accuracy: 0.9796
Epoch 90/400
35/46 [=====================>........] - ETA: 0s - loss: 0.0278 - accuracy: 0.9786
Yes You are here 0.9789115786552429 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:29        17192
config.json                                    2023-05-15 07:52:29         1588
metadata.json                                  2023-05-15 07:52:29           64
46/46 [================

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:30        17192
config.json                                    2023-05-15 07:52:30         1588
metadata.json                                  2023-05-15 07:52:30           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0260 - accuracy: 0.9776 - val_loss: 0.0984 - val_accuracy: 0.9796
Epoch 100/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0255 - accuracy: 0.9796 - val_loss: 0.0979 - val_accuracy: 0.9796
Epoch 101/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0246 - accuracy: 0.9839
Yes You are here 0.9782313108444214 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
........

metadata.json                                  2023-05-15 07:52:31           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0243 - accuracy: 0.9837 - val_loss: 0.0959 - val_accuracy: 0.9796
Epoch 109/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0257 - accuracy: 0.9748 - val_loss: 0.1016 - val_accuracy: 0.9864
Epoch 110/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0244 - accuracy: 0.9865
Yes You are here 0.9836734533309937 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                           

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:34        17192
config.json                                    2023-05-15 07:52:34         1588
metadata.json                                  2023-05-15 07:52:34           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0204 - accuracy: 0.9857 - val_loss: 0.1022 - val_accuracy: 0.9864
Epoch 130/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0207 - accuracy: 0.9816 - val_loss: 0.1023 - val_accuracy: 0.9796
Epoch 131/400
32/46 [===================>..........] - ETA: 0s - loss: 0.0192 - accuracy: 0.9834
Yes You are here 0.9782313108444214 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
........

46/46 [==============================] - 0s 3ms/step - loss: 0.0173 - accuracy: 0.9857 - val_loss: 0.1045 - val_accuracy: 0.9796
Epoch 150/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0178 - accuracy: 0.9837 - val_loss: 0.1066 - val_accuracy: 0.9796
Epoch 151/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0175 - accuracy: 0.9857 - val_loss: 0.1020 - val_accuracy: 0.9796
Epoch 152/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0182 - accuracy: 0.9816 - val_loss: 0.1029 - val_accuracy: 0.9796
Epoch 153/400
43/46 [===========================>..] - ETA: 0s - loss: 0.0176 - accuracy: 0.9847
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
....

variables.h5                                   2023-05-15 07:52:39        17192
config.json                                    2023-05-15 07:52:39         1588
metadata.json                                  2023-05-15 07:52:39           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0171 - accuracy: 0.9816 - val_loss: 0.1068 - val_accuracy: 0.9864
Epoch 174/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0164 - accuracy: 0.9864 - val_loss: 0.1062 - val_accuracy: 0.9796
Epoch 175/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0160 - accuracy: 0.9864 - val_loss: 0.1065 - val_accuracy: 0.9796
Epoch 176/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0164 - accuracy: 0.9837 - val_loss: 0.1066 - val_accuracy: 0.9796
Epoch 177/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0151 - accuracy: 0.9837 - val_loss: 0.1066 - val_accuracy: 0.9796
Epoch 178/400
38/46 [=======================>......] - ETA:

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:41        17192
config.json                                    2023-05-15 07:52:41         1588
metadata.json                                  2023-05-15 07:52:41           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0156 - accuracy: 0.9844 - val_loss: 0.1052 - val_accuracy: 0.9864
Epoch 193/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0144 - accuracy: 0.9878 - val_loss: 0.1043 - val_accuracy: 0.9728
Epoch 194/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0150 - accuracy: 0.9823 - val_loss: 0.1088 - val_accuracy: 0.9796
Epoch 195/400
28/46 [=================>............] - ETA: 0s - loss: 0.0155 - accuracy: 0.9866
Yes You are here 0.9843537211418152 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
...........

variables.h5                                   2023-05-15 07:52:43        17192
config.json                                    2023-05-15 07:52:43         1588
metadata.json                                  2023-05-15 07:52:43           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0148 - accuracy: 0.9823 - val_loss: 0.1054 - val_accuracy: 0.9864
Epoch 209/400
33/46 [====================>.........] - ETA: 0s - loss: 0.0131 - accuracy: 0.9886
Yes You are here 0.9877551198005676 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5          

metadata.json                                  2023-05-15 07:52:44           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0132 - accuracy: 0.9878 - val_loss: 0.1044 - val_accuracy: 0.9864
Epoch 219/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0131 - accuracy: 0.9905 - val_loss: 0.1053 - val_accuracy: 0.9728
Epoch 220/400
43/46 [===========================>..] - ETA: 0s - loss: 0.0139 - accuracy: 0.9847
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                           

metadata.json                                  2023-05-15 07:52:46           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0133 - accuracy: 0.9857 - val_loss: 0.1034 - val_accuracy: 0.9864
Epoch 230/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0129 - accuracy: 0.9864 - val_loss: 0.0974 - val_accuracy: 0.9796
Epoch 231/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0138 - accuracy: 0.9875
Yes You are here 0.9870748519897461 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                           

metadata.json                                  2023-05-15 07:52:47           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0134 - accuracy: 0.9871 - val_loss: 0.1046 - val_accuracy: 0.9864
Epoch 241/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0120 - accuracy: 0.9905 - val_loss: 0.1013 - val_accuracy: 0.9796
Epoch 242/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0125 - accuracy: 0.9864 - val_loss: 0.1069 - val_accuracy: 0.9864
Epoch 243/400
38/46 [=======================>......] - ETA: 0s - loss: 0.0129 - accuracy: 0.9893
Yes You are here 0.9891156554222107 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Kera

variables.h5                                   2023-05-15 07:52:49        17192
config.json                                    2023-05-15 07:52:49         1588
metadata.json                                  2023-05-15 07:52:49           64
46/46 [==============================] - 0s 2ms/step - loss: 0.0121 - accuracy: 0.9905 - val_loss: 0.1041 - val_accuracy: 0.9864
Epoch 253/400
35/46 [=====================>........] - ETA: 0s - loss: 0.0117 - accuracy: 0.9848
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5          

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:50        17192
config.json                                    2023-05-15 07:52:50         1588
metadata.json                                  2023-05-15 07:52:50           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0117 - accuracy: 0.9884 - val_loss: 0.1061 - val_accuracy: 0.9864
Epoch 266/400
29/46 [=================>............] - ETA: 0s - loss: 0.0108 - accuracy: 0.9860
Yes You are here 0.9870748519897461 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:52        17192
config.json                                    2023-05-15 07:52:52         1588
metadata.json                                  2023-05-15 07:52:52           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0112 - accuracy: 0.9912 - val_loss: 0.1047 - val_accuracy: 0.9864
Epoch 278/400
24/46 [==============>...............] - ETA: 0s - loss: 0.0108 - accuracy: 0.9909
Yes You are here 0.9911564588546753 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3


Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:53        17192
config.json                                    2023-05-15 07:52:53         1588
metadata.json                                  2023-05-15 07:52:53           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0105 - accuracy: 0.9912 - val_loss: 0.1018 - val_accuracy: 0.9864
Epoch 291/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0106 - accuracy: 0.9918 - val_loss: 0.1018 - val_accuracy: 0.9796
Epoch 292/400
33/46 [==========

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:55        17192
config.json                                    2023-05-15 07:52:55         1588
metadata.json                                  2023-05-15 07:52:55           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0105 - accuracy: 0.9918 - val_loss: 0.1038 - val_accuracy: 0.9864
Epoch 301/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0106 - accuracy: 0.9889
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

Epoch 309/400
37/46 [=======================>......] - ETA: 0s - loss: 0.0119 - accuracy: 0.9882
Yes You are here 0.9884353876113892 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:56        17192
config.json                                    2023-05-15 07:52:56         1588
metadata.json                                  2023-05-15 07:52:56           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0118 - accuracy: 0.9884 - val_loss: 0.1016 - val_accuracy: 0.9864
Epoch 310/400
30/46 [=

33/46 [====================>.........] - ETA: 0s - loss: 0.0111 - accuracy: 0.9896
Yes You are here 0.9884353876113892 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:58        17192
config.json                                    2023-05-15 07:52:58         1588
metadata.json                                  2023-05-15 07:52:58           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0110 - accuracy: 0.9884 - val_loss: 0.0980 - val_accuracy: 0.9864
Epoch 320/400
34/46 [===============

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:52:59        17192
config.json                                    2023-05-15 07:52:59         1588
metadata.json                                  2023-05-15 07:52:59           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0102 - accuracy: 0.9905 - val_loss: 0.0994 - val_accuracy: 0.9864
Epoch 328/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0097 - accuracy: 0.9919
Yes You are here 0.9897959232330322 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

metadata.json                                  2023-05-15 07:53:00           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0092 - accuracy: 0.9932 - val_loss: 0.0994 - val_accuracy: 0.9864
Epoch 337/400
38/46 [=======================>......] - ETA: 0s - loss: 0.0093 - accuracy: 0.9901
Yes You are here 0.9891156554222107 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:00        17192
config.json                                    2023-05-15 07:53:00         1588
metadata.json         

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:01        17192
config.json                                    2023-05-15 07:53:01         1588
metadata.json                                  2023-05-15 07:53:01           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0095 - accuracy: 0.9912 - val_loss: 0.0937 - val_accuracy: 0.9864
Epoch 346/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0097 - accuracy: 0.9896
Yes You are here 0.9918367266654968 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

config.json                                    2023-05-15 07:53:03         1588
metadata.json                                  2023-05-15 07:53:03           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0097 - accuracy: 0.9912 - val_loss: 0.0943 - val_accuracy: 0.9864
Epoch 356/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0087 - accuracy: 0.9899
Yes You are here 0.9911564588546753 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:03        17192
config.json           

Epoch 365/400
37/46 [=======================>......] - ETA: 0s - loss: 0.0100 - accuracy: 0.9916
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:04        17192
config.json                                    2023-05-15 07:53:04         1588
metadata.json                                  2023-05-15 07:53:04           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0100 - accuracy: 0.9905 - val_loss: 0.0974 - val_accuracy: 0.9864
Epoch 366/400
43/46 [=

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:05        17192
config.json                                    2023-05-15 07:53:05         1588
metadata.json                                  2023-05-15 07:53:05           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0090 - accuracy: 0.9912 - val_loss: 0.0952 - val_accuracy: 0.9864
Epoch 374/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0099 - accuracy: 0.9891 - val_loss: 0.0994 - val_accuracy: 0.9796
Epoch 375/400
32/46 [===================>..........] - ETA: 0s - loss: 0.0088 - accuracy: 0.9893
Yes You are here 0.9897959232330322 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
........

config.json                                    2023-05-15 07:53:07         1588
metadata.json                                  2023-05-15 07:53:07           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0085 - accuracy: 0.9918 - val_loss: 0.0942 - val_accuracy: 0.9864
Epoch 384/400
33/46 [====================>.........] - ETA: 0s - loss: 0.0082 - accuracy: 0.9915
Yes You are here 0.9918367266654968 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:07        17192
config.json           

28/46 [=================>............] - ETA: 0s - loss: 0.0079 - accuracy: 0.9933
Yes You are here 0.9911564588546753 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:08        17192
config.json                                    2023-05-15 07:53:08         1588
metadata.json                                  2023-05-15 07:53:08           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0086 - accuracy: 0.9912 - val_loss: 0.0947 - val_accuracy: 0.9864
Epoch 393/400
38/46 [===============

File Name                                             Modified             Size
variables.h5                                   2023-05-15 07:53:10        17192
config.json                                    2023-05-15 07:53:10         1588
metadata.json                                  2023-05-15 07:53:10           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0085 - accuracy: 0.9912 - val_loss: 0.0938 - val_accuracy: 0.9864


In [24]:
accurate_train_accuracy,classify_report= test_model(model,X_train,Y_train)

46/46 [==============================] - 0s 904us/step


In [25]:
accurate_train_accuracy

0.9931972789115646

In [26]:
print(classify_report)

              precision    recall  f1-score   support

           0       0.97      1.00      0.99       272
           1       0.99      0.97      0.98       278
           2       1.00      0.99      1.00       376
           3       1.00      1.00      1.00       544

    accuracy                           0.99      1470
   macro avg       0.99      0.99      0.99      1470
weighted avg       0.99      0.99      0.99      1470



In [27]:
accurate_val_accuracy,classify_report= test_model(model,X_val,Y_val)

5/5 [==============================] - 0s 1ms/step


In [28]:
accurate_val_accuracy

0.9863945578231292

In [29]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      0.95      0.97        20
           1       0.95      1.00      0.98        20
           2       0.97      1.00      0.99        36
           3       1.00      0.99      0.99        71

    accuracy                           0.99       147
   macro avg       0.98      0.98      0.98       147
weighted avg       0.99      0.99      0.99       147



In [30]:
accurate_test_accuracy,classify_report= test_model(model,X_test,Y_test)

5/5 [==============================] - 0s 1ms/step


In [31]:
accurate_test_accuracy

0.9932432432432432

In [32]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        18
           1       1.00      1.00      1.00        20
           2       1.00      0.98      0.99        45
           3       0.98      1.00      0.99        65

    accuracy                           0.99       148
   macro avg       1.00      0.99      1.00       148
weighted avg       0.99      0.99      0.99       148



## Notes

- name            : first model 
- split           : 1/3 
- oversampling    : 0.5 
- val_acc         : 0.9836734693877551
- test_acc        : 0.983739837398374
- gamma           : 1 
---------------------------------------------------------------------------------------------
- name            : second model 
- split           : 1/3 
- oversampling    : 0.8
- val_acc         : 0.9877551020408163
- test_acc        : 0.9878048780487805
- gamma           : 1 

---------------------------------------------------------------------------------------------

- name            : forth model 
- split           : 1/5
- oversampling    : 1
- val_acc         : 0.9863945578231292
- test_acc        : 0.9932432432432432
- gamma           : 2

---------------------------------------------------------------------------------------------
- name            : fifth model 
- split           : 1/5
- oversampling    : 0.5
- val_acc         : 0.9863945578231292
- test_acc        : 0.9864864864864865
- gamma           : 2

---------------------------------------------------------------------------------------------

